### DISCUSSION TITLE
Experimental macro

### DISCUSSION THREAD

**Discussion by johanholmerin** [adelsz/pgtyped#393] [Show and tell]
Inspired by [sqlx](https://github.com/launchbadge/sqlx), and to avoid file generation, I created a library for [type-safe macros in TypeScript](https://github.com/johanholmerin/tscm) and a [pgtyped macro](https://github.com/johanholmerin/tscm-examples/tree/master/macros/pgtyped). Not production ready, but may still be of interest.

https://user-images.githubusercontent.com/7433263/150874144-6a3e663b-52db-4c5e-ad7d-6a37946bf8f0.mp4

**Comment by adelsz** [comment]
Cool stuff! I wish if TS offered a proper macro escape hatch or custom AST rewriterer like ReasonMLs PPXs.
